### ISSUE TITLE
Custom Gradient Presets: Calculate default control points if none are provided

### ISSUE THREAD

**Issue by kjellr** [WordPress/gutenberg#27382]
If a preset gradient does not explicitly include positions for each included color value, the position controls do not appear in the UI (Also, editing options can cause the block to crash when this is the case).

Gutenberg should be able to calculate the color stop defaults. 

For example...
`linear-gradient(160deg, cyan, orange)` 
... should be processed by gutenberg to be ... 
`linear-gradient(160deg, cyan 0%, orange 100%)`. 

## Testing 

1. Add a gradient preset to a theme, but do not specify the color stop positions. (e.g. `linear-gradient(160deg, cyan, orange)`).
2. Assign that gradient to a block.
3. Try to modify the type of gradient (Linear/Radial). Note that this has no effect.
5. Try to modify the gradient angle. Note that the block crashes. 

Thanks to some sleuthing by @ryelle, the issue appears related to Gutenberg not entirely understanding the preset: the color stops are not being added into the Gutenberg control. When the preset is modified to include those stops, the issue goes away. 

## Screenshot

![Kapture 2020-11-30 at 15 58 45](https://user-images.githubusercontent.com/1202812/100664022-168bcf80-3325-11eb-9f0a-0e4704cd07ac.gif)

_Related:_ https://github.com/WordPress/twentytwentyone/pull/904/
_Tested with Gutenberg 9.4.1_

**Comment by stokesman** [comment]
I looked into this some time ago found a solution. I think I got derailed on making a PR when I was considering how other gradient rules might be supported. Looking at it again, I think it's probably fine to just support the basic omission of stop positions as in the example in this issue.

So now I've made #27413, which I've yet to add description for just wanted to get a note in about it here.

Also, this is the same as #23501. Maybe one should get closed as a duplicate.
